python/lsst/cp/pipe/linearity.py

```diff
@@ -0,0 +1,235 @@
+# This file is part of cp_pipe.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+import numpy as np
+
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+import lsst.pex.config as pexConfig
+
+from lsstDebug import getDebugFrame
+from lsst.ip.isr import (Linearizer, IsrProvenance)
+
+from .utils import (fitLeastSq, funcPolynomial)
+
+
+__all__ = ["LinearitySolveTask", "LinearitySolveConfig"]
+
+
+class LinearitySolveConnections(pipeBase.PipelineTaskConnections,
+                                dimensions=("instrument", "detector")):
+
+    inputPtc = cT.Input(
+        name="inputPtc",
+        doc="Input PTC dataset.",
+        storageClass="StructuredDataDict",
+        dimensions=("instrument", "detector"),
+        multiple=False,
+    )
+    camera = cT.Input(
+        name="camera",
+        doc="Camera Geometry definition.",
+        storageClass="Camera",
+        dimensions=("instrument", ),
+    )
+    outputLinearizer = cT.Output(
+        name="linearity",
+        doc="Output linearity measurements.",
+        storageClass="Linearizer",
+        dimensions=("instrument", "detector"),
+    )
+
+
+class LinearitySolveConfig(pipeBase.PipelineTaskConfig,
+                           pipelineConnections=LinearitySolveConnections):
+    """Configuration for solving the linearity from PTC dataset.
+    """
+    linearityType = pexConfig.ChoiceField(
+        dtype=str,
+        doc="Type of linearizer to construct.",
+        default="Polynomial",
+        allowed={
+            "LookupTable": "Create a lookup table solution.",
+            "Polynomial": "Create an arbitrary polynomial solution.",
+            "Squared": "Create a single order squared solution.",
+            "None": "Create a dummy solution.",
```

Comment:
Craig has recently been obtaining results that indicate that we might also need to implement a spline fit option. Could this be something added to this ticket, or perhaps something for another ticket?

Reply:
Another ticket please.

Reply:
I've added DM-26545 for this.